app/services/technology_growth_mom_calculator.rb

```diff
@@ -0,0 +1,45 @@
+class TechnologyGrowthMomCalculator < BaseService
+  def initialize(metric_name, months)
+    @metric_name = metric_name
+    @months = months
+  end
+
+  def call
+    Technology.all.map do |technology|
+      {
+        name: technology.name.titlecase,
+        data: growth_mom_for(technology)
+      }
+    end
+  end
+
+  private
+
+  attr_reader :metric_name, :months
+
+  def growth_mom_for(technology)
+    metrics = metrics_for(technology)
+
+    metrics.each.with_index.inject({}) do |hash, metric_with_index|
+      metric, index = metric_with_index
+      previous_metric = metrics[index - 1]
+      next(hash) if index.zero? || previous_metric.value.zero?
```

Comment:
Would

```suggestion
      next if index.zero? || previous_metric.value.zero?
```

work too?

Reply:
It wouldn't, I've tried it. It seems that it needs to keep passing the seed on to the next iteration.